brainglobe_template_builder/preproc/brightness.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import SimpleITK as sitk
+
+
+def correct_image_brightness(
+    input_image: np.ndarray, spacing: list[float]
+) -> np.ndarray:
+    """Correct image brightness using simple itk's
+    N4BiasFieldCorrectionImageFilter.
+
+    Default values are set to match those used in
+    ants.n4_bias_field_correction.
+
+    Parameters
+    ----------
+    input_image : np.ndarray
+        Input image to process.
+    spacing : list
+        Pixel size (in mm) for each axis - should
+        be in same order as input_image axes.
+
+    Returns
+    ----------
+    np.ndarray
+        The processed image with float64 dtype.
+    """
+
+    # The output of GetLogBiasFieldAsImage is always float32,
+    # so we need a float32 input image to avoid errors
+    if input_image.dtype != np.float32:
+        input_image = input_image.astype(np.float32, casting="same_kind")
+
+    sitk_image = sitk.GetImageFromArray(input_image)
+
+    # Conversion to sitk flips the axis order [z, y, x] -> [x, y, z],
+    # so we must flip the spacing too
+    sitk_image.SetSpacing(spacing[::-1])
```

Comment:
~~Discussing this with @niksirbi - we assume that the `input_image` array here is in ASR, right?
Because not all re-orderings of axes have the same map to the default axis ordering that `sitk.GetImageFromArray` assumes?~~

Actually, sorry @K-Meech - we think this is fine after all :joy: